pkgs/development/compilers/julia/1.0.nix

```diff
@@ -1,211 +1,154 @@
+{
+fetchzip, fetchFromGitHub, lib, pkgs, stdenv
+# Build tools
+, cmake, perl, python3, which
+# Dependencies
+, arpack, curl, blas, gmp, lapack, libssh2, libunwind, mpfr, pcre2
+# Testing
+, openssl
+# Darwin
+, ApplicationServices, CoreServices
 }:
 
+# Ensure dependencies from Nixpkgs are equal or greater to vendored versions.
+assert lib.versionAtLeast (lib.strings.getVersion gmp) "6.1.2";
+assert lib.versionAtLeast arpack.version "3.3.0";
+assert lib.versionAtLeast curl.version "7.56.0";
+assert lib.versionAtLeast libssh2.version "1.8.2";
+assert lib.versionAtLeast libunwind.version "1.1";
+assert lib.versionAtLeast mpfr.version "4.0.2";
+assert lib.versionAtLeast pcre2.version "10.30";
+# Can not assert for BLAS and LAPACK due to multiple providers, Julia uses:
+#   * OpenBLAS: e8a68ef261a33568b0f0cf53e0e2287e9f12e69e
+#   * LAPACK: 3.5.0
+
+# Ensure BLAS and Lapack integer size compatibility.
+assert (blas.isILP64 == lapack.isILP64);
 
 let
-  majorVersion = "1";
-  minorVersion = "0";
-  maintenanceVersion = "4";
-  src_sha256 = "1dfx68wbrrzpbh74rla7i2g3r5z6wa1pxq3ahyfm5m27vfyjbkhg";
-
-  libuvVersion = "ed3700c849289ed01fe04273a7bf865340b2bd7e";
-  libuvSha256 = "137w666zsjw1p0ma3lf94d75hr1q45sgkfmbizkyji2qm57cnxjs";
-
-  dsfmtVersion = "2.2.3";
-  dsfmt = fetchurl {
-    url = "http://www.math.sci.hiroshima-u.ac.jp/~m-mat/MT/SFMT/dSFMT-src-${dsfmtVersion}.tar.gz";
-    sha256 = "03kaqbjbi6viz0n33dk5jlf6ayxqlsq4804n7kwkndiga9s4hd42";
-  };
-
-  libuv = fetchurl {
-    url = "https://api.github.com/repos/JuliaLang/libuv/tarball/${libuvVersion}";
-    sha256 = libuvSha256;
-  };
-
-  rmathVersion = "0.1";
-  rmath-julia = fetchurl {
-    url = "https://api.github.com/repos/JuliaLang/Rmath-julia/tarball/v${rmathVersion}";
-    sha256 = "1qyps217175qhid46l8f5i1v8i82slgp23ia63x2hzxwfmx8617p";
-  };
-
-  virtualenvVersion = "15.0.0";
-  virtualenv = fetchurl {
-    url = "mirror://pypi/v/virtualenv/virtualenv-${virtualenvVersion}.tar.gz";
-    sha256 = "06fw4liazpx5vf3am45q2pdiwrv0id7ckv7n6zmpml29x6vkzmkh";
-  };
-
-  libwhichVersion = "81e9723c0273d78493dc8c8ed570f68d9ce7e89e";
-  libwhich = fetchurl {
-    url = "https://api.github.com/repos/vtjnash/libwhich/tarball/${libwhichVersion}";
-    sha256 = "1p7zg31kpmpbmh1znrk1xrbd074agx13b9q4dcw8n2zrwwdlbz3b";
-  };
-
-  llvmVersion = "6.0.0";
-  llvm = fetchurl {
-    url = "http://releases.llvm.org/6.0.0/llvm-${llvmVersion}.src.tar.xz";
-    sha256 = "0224xvfg6h40y5lrbnb9qaq3grmdc5rg00xq03s1wxjfbf8krx8z";
-  };
-
-  suitesparseVersion = "4.4.5";
-  suitesparse = fetchurl {
-    url = "http://faculty.cse.tamu.edu/davis/SuiteSparse/SuiteSparse-${suitesparseVersion}.tar.gz";
-    sha256 = "1jcbxb8jx5wlcixzf6n5dca2rcfx6mlcms1k2rl5gp67ay3bix43";
-  };
-  version = "${majorVersion}.${minorVersion}.${maintenanceVersion}";
-in
-
-stdenv.mkDerivation rec {
+  # Pre-1.6.x Julia is not compatible with libgit2 1.x, use an earlier version.
+  libgit2 = pkgs.libgit2.overrideAttrs (oldAttrs: rec {
```

Comment:
```suggestion
  libgit2 = libgit2.overrideAttrs (oldAttrs: rec {
```

Reply:
How much is this style frowned upon? My preference for it is in that it avoids shadowing `libgit2` inside the expression and thus it is less likely that you accidentally refer to the non-overridden version. I am happy to conform with standards though if it really stings your eyes.

Reply:
I think this breaks splicing in non-obvious ways.

Reply:
Not sure if I can parse “splicing”, any references or advice would be welcome. Is it as easy falling back on shadowing and it should work as intended?

Reply:
Splicing is the dark magic that ensures packages used in nativeBuildInputs are compiled for the build machine whereas buildInputs are compiled for the host machine.

Source for "it breaks splicing": #117053 Well actually it does not break it, but it has complex, unintended effects.

I think shadowing is fine. You can also do `let my_awesome_libgit2 = ... in ... my_awesome_libgit2`

Reply:
Roger that, I think I understand how it could happen although not on a lower level. Thank you for educating me. I have accommodated the changes and pushed a new version.